### PR TITLE
feat(insured-bridge): Add ArbitrumMessenger

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

`BridgeAdmin` should be able to relay messages to Arbitrum L2.

**Summary**

Implement `ArbitrumMessenger`'s `relayMessage` function.

**Details**

This may be unnecessary for some PRs. Catch-all for detailed explanations about the implementation decisions and implications of the change.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import "./iAVM_Inbox.sol";
+
+abstract contract AVM_CrossDomainEnabled {
+    modifier onlyFromCrossDomainAccount(address l1Counterpart) {
+        require(msg.sender == applyL1ToL2Alias(l1Counterpart), "ONLY_COUNTERPART_GATEWAY");
+        _;
+    }
+
+    uint160 constant offset = uint160(0x1111000000000000000000000000000000001111);
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol:11]: nit: we should consider having a common const file for these things. this is repeated from here: https://github.com/UMAprotocol/protocol/blob/7d73fde9cf062c50a599166a0fe52f3f2a817ddb/packages/core/contracts-ovm/insured-bridge/external/arbitrum/AVM_CrossDomainEnabled.sol#L9

dont need to do this in this PR though.

File: packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * @param _inbox Contract that sends generalized messages to the Arbitrum chain.
+     */
+    constructor(address _inbox) {
+        inbox = iAVM_Inbox(_inbox);
+    }
+
+    // l1 addresses are transformed during l1->l2 calls. see https://developer.offchainlabs.com/docs/l1_l2_messages#address-aliasing for more information.
+    function applyL1ToL2Alias(address l1Address) internal pure returns (address l2Address) {
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol:22]: same as for this function

Comment (chrismaree) [reply packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol:22]: actually why are you not re-using this file? https://github.com/UMAprotocol/protocol/blob/7d73fde9cf062c50a599166a0fe52f3f2a817ddb/packages/core/contracts-ovm/insured-bridge/external/arbitrum/AVM_CrossDomainEnabled.sol#L3

you simply need to add the one additional function. much is the same, no?

File: packages/core/contracts-ovm/insured-bridge/external/iAVM_Inbox.sol [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Copied mostly from https://github.com/makerdao/arbitrum-dai-bridge/blob/7f1b47ef65a43f1696c5f1681109daac127d9c95/contracts/arbitrum/IInbox.sol
+// - Bumped solidity version to >= 0.8.x
+// - Removed functions from interface that are not used by AVM_CrossDomainEnabled.sol
+// - Removed IMessageProvider inheritance.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2021, Offchain Labs, Inc.
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/external/iAVM_Inbox.sol:8]: do we need to include this license? if you reference where it came from do we need this bit?

File: packages/core/test/insured-bridge/ArbitrumMessenger.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    l2Token = rando2;
+    await runDefaultFixture(hre);
+    timer = await Timer.deployed();
+    finder = await Finder.deployed();
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    collateralWhitelist = await AddressWhitelist.deployed();
+    await identifierWhitelist.methods.addSupportedIdentifier(defaultIdentifier).send({ from: owner });
+  });
+  beforeEach(async function () {
+    l1InboxMock = await deployContractMock("AVM_InboxMock", {}, AVM_InboxMock);
```
Comment (chrismaree) [inline packages/core/test/insured-bridge/ArbitrumMessenger.js:60]: perfect use of this. I'd like to refactor the `deployContractMock` in the future to make the params a bit cleaner but for now it's fine

File: packages/core/contracts-ovm/insured-bridge/implementation/ArbitrumMessenger.sol [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * @notice Sends a message to an account on L2.
+     * @param target The intended recipient on L2.
+     * @param gasLimit The gasLimit for the receipt of the message on L2.
+     * @param gasPrice Gas price bid for L2 execution.
+     * @param message The data to send to the target (usually calldata to a function with
+     *  `onlyFromCrossDomainAccount()`)
+     */
+    function relayMessage(
```
Comment (chrismaree) [inline packages/core/contracts-ovm/insured-bridge/implementation/ArbitrumMessenger.sol:24]: 🧼

**Review by chrismaree — APPROVED** [review]
This is great!

other than the duplication of `AVM_CrossDomainEnabled` this is perfect

File: packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol (hunk 1 above)
Comment (nicholaspai) [reply packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol:11]: Ah I had made a local copy of `AVM_CrossDomainEnabled` to get my tests to work, will merge with your PR and remove duplicate

File: packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol (hunk 2 above)
Comment (nicholaspai) [reply packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol:22]: See above response

File: packages/core/contracts-ovm/insured-bridge/external/iAVM_Inbox.sol (hunk 3 above)
Comment (nicholaspai) [reply packages/core/contracts-ovm/insured-bridge/external/iAVM_Inbox.sol:8]: Removed

**Pull request merged**